site-scrapers/BostonMedicalCenter/index.js

```diff
@@ -0,0 +1,57 @@
+const { sites } = require("./config");
+const mychart = require("../../lib/MyChartAPI");
+
+module.exports = async function GetAvailableAppointments() {
+    console.log("BMC starting.");
+    const webData = await ScrapeWebsiteData();
+    console.log("BMC done.");
+    const results = [];
+    const timestamp = new Date();
+
+    for (const site of sites) {
+        results.push({
+            ...webData[site.departmentID],
+            ...site,
+            timestamp,
+        });
+    }
+    return results;
+};
+
+async function ScrapeWebsiteData() {
+    // We need to go through the flow and use a request verification token
+    const [
+        cookie,
+        verificationToken,
+    ] = await mychart.GetCookieAndVerificationToken(
+        "https://mychartscheduling.bmc.org/mychartscheduling/SignupAndSchedule/EmbeddedSchedule?id=10033319,10033364,10033367,10033370,10033373&dept=10098245,10098242,10098243,10098244,10098241&vt=2008&lang=en-US"
+    );
+    return mychart.AddFutureWeeks(
+        "mychartscheduling.bmc.org",
+        "/MyChartscheduling/OpenScheduling/OpenScheduling/GetOpeningsForProvider?noCache=0.4024598146273777",
+        cookie,
+        verificationToken,
+        10,
+        PostDataCallback
+    );
+}
+
+/**
+ * This is the callback function
+ */
+function PostDataCallback(startDateFormatted) {
```

Comment:
The comment is ... not helping us out here.
I accept that `LynnTech` and `SouthBostonCHC` started us down this path, so maybe we should neaten them up as well.